diff --git a/stmmac_main.c.orig b/stmmac_main.c
index 9c69edc20..3d178601d 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -2299,11 +2299,11 @@ static int stmmac_init_dma_engine(struct stmmac_priv *priv)
 	if (priv->extend_desc && (priv->mode == STMMAC_RING_MODE))
 		atds = 1;
 
-	ret = stmmac_reset(priv, priv->ioaddr);
-	if (ret) {
-		dev_err(priv->device, "Failed to reset the dma\n");
-		return ret;
-	}
+//	ret = stmmac_reset(priv, priv->ioaddr);
+//	if (ret) {
+//		dev_err(priv->device, "Failed to reset the dma\n");
+//		return ret;
+//	}
 
 	/* DMA Configuration */
 	stmmac_dma_init(priv, priv->ioaddr, priv->plat->dma_cfg, atds);
